test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/TestAutoCreateSharedArchiveUpgrade.java

```diff
@@ -134,11 +164,75 @@ static void assertCreatedJSA(OutputAnalyzer output) {
         output.shouldContain("Dumping shared data to file");
     }
 
-    static void assertJSAVersionMismatch(OutputAnalyzer output) {
-        output.shouldContain("does not match the required version");
-    }
-
     static void assertUsedJSA(OutputAnalyzer output) {
         output.shouldContain("Mapped dynamic region #0");
     }
+
+    // Fetch JDK artifact depending on platform
+    // If the artifact cannot be found, default to the test.boot.jdk property
+    private static String fetchJDK(int version) throws Throwable {
+        int build;
+        String architecture;
+        HashMap<String, Object> jdkArtifactMap = new HashMap<>();
+        jdkArtifactMap.put("server", "jpg");
+        jdkArtifactMap.put("product", "jdk");
+
+        // Select the correct release build number for each version
+        // *UPDATE THIS* after each release
+        switch(version) {
+            case 19:
+                build = 36;
+                break;
+            case 20:
+                build = 29;
+                break;
+            default:
+                throw new RuntimeException("Unsupported JDK version " + version);
+        }
+        jdkArtifactMap.put("version", version);
+        jdkArtifactMap.put("build_number", build);
+
+        // Get correct file name for architecture
+        if (Platform.isX64()) {
+            architecture = "x";
+        } else if (Platform.isAArch64()) {
+            architecture = "aarch";
```

Comment:
It's better to use "x64" and "aarch64" here. The comments below should be fixed, too.